packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,1220 @@
+        // and update the state when the promise returns
+        let promise: PromiseLike<ISelectableOption[]> = this._currentPromise = newOptions;
+        promise.then((newOptionsFromPromise: ISelectableOption[]) => {
+          if (promise === this._currentPromise) {
+            this.setState({
+              currentOptions: newOptionsFromPromise
+            });
+          }
+        });
+      }
+    }
+  }
+
+  /**
+   * OnBlur handler. Set the focused state to false
+   * and submit any pending value
+   */
+  @autobind
+  private _onBlur() {
+    if (this.state.focused) {
+      this.setState({ focused: false });
+      this._submitPendingValue();
+    }
+  }
+
+  /**
+   * Submit a pending value if there is one
+   */
+  private _submitPendingValue() {
+    let {
+      onChanged,
+      allowFreeform
+    } = this.props;
+    let {
+      currentPendingValue,
+      currentPendingValueValidIndex,
+      currentOptions
+    } = this.state;
+
+    // If we allow freeform and we have a pending value, we
+    // need to handle that
+    if (allowFreeform && currentPendingValue !== '') {
+
+      // Check to see if the user typed an exact match
+      if (currentPendingValueValidIndex >= 0) {
+        let pendingOptionText: string = currentOptions[currentPendingValueValidIndex].text.toLocaleLowerCase();
+
+        // By exact match, that means: our pending value is the same as the the pending option text OR
+        // the peding option starts with the pending value and we have an "autoComplete" selection
+        // where the total lenght is equal to pending option length; update the state
+        if (currentPendingValue.toLocaleLowerCase() === pendingOptionText ||
+          (pendingOptionText.indexOf(currentPendingValue.toLocaleLowerCase()) === 0 &&
+            this._comboBox.isValueSelected &&
+            currentPendingValue.length + (this._comboBox.selectionEnd - this._comboBox.selectionStart) === pendingOptionText.length)) {
+          this._setSelectedIndex(currentPendingValueValidIndex);
+          this._clearPendingInfo();
+          return;
+        }
+      }
+
+      // Create a new option
+      let newOption: ISelectableOption = { key: currentPendingValue, text: currentPendingValue };
+      let newOptions: ISelectableOption[] = [...currentOptions, newOption];
+      let newSelectedIndex: number = this._getSelectedIndex(newOptions, currentPendingValue);
+
+      this.setState({
+        currentOptions: newOptions,
+        selectedIndex: newSelectedIndex
+      });
+
+      if (onChanged) {
+        onChanged(null, null, currentPendingValue);
+      }
+    } else if (currentPendingValueValidIndex >= 0) {
+      // Since we are not allowing freeform, we must have a matching
+      // to be able to update state
+      this._setSelectedIndex(currentPendingValueValidIndex);
+    }
+
+    // Finally, clear the pending info
+    this._clearPendingInfo();
+  }
+
+  // Render Callout container and pass in list
+  @autobind
```

Comment:
Yes they do need autobind since they reference "this" which won't be the expected this. I had narrowed down the autobind usages already every one that's there now needs to be there and would throw an error otherwise